File: Data Structures & Algorithms/duplicate-integer/submission-2.py
class Solution:
    def hasDuplicate(self, nums: List[int]) -> bool:
        s = set()

        for num in nums:
            if num in s:
                return True
            else:
                s.add(num)
        
        return False



        # s = set(nums)
        # return len(nums) != len(s)


        # c = Counter(nums)
        # for key, val in c.items():
        #     if val > 1:
        #         return True
        
        # return False

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-2.cpp
class Solution {
   public:
    int longestConsecutive(vector<int>& nums) {
        if(nums.size()==0)return 0;
        map<int, int> mp;
        for (auto num : nums) {
            mp[num]++;
        }

        vector<int> res;
        for (auto m : mp) {
            res.push_back(m.first);
        }


        if (res.size() == 1) return 1;
        int ans = 1;
        int i = 0;
        for (int j = 1; j < res.size(); j++) {
            if (res[j] - res[j - 1] == 1) {
                ans = max(ans, j - i+1);

            } else {
                i = j;
            }
        }
        return ans;
    }
};
